File: ruby-gem/lib/models/labelling_characteristic.rb
# frozen_string_literal: true

# Class from DFC Business Ontology: #LabellingCharacteristic
require_relative '../core/semantic_object'


module DfcLinkmlConnector
  module Models
    class LabellingCharacteristic < Core::SemanticObject
      SEMANTIC_TYPE = "dfc-b:LabellingCharacteristic".freeze

      # @return [String]
      attr_accessor :labelling_dimension

      # @return [String]
      attr_accessor :labelling_characteristic_of

      # @return [String]
      attr_accessor :date

      # @return [String]
      attr_accessor :description

      # @return [String]
      attr_accessor :name

      # @return [String]
      attr_accessor :characteristic_of

      # @return [String]
      attr_accessor :dimension

      # @param semanticId [String]
      # @param labellingDimension: nil, labellingCharacteristicOf: nil, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil
      def initialize(semanticId, labellingDimension: nil, labellingCharacteristicOf: nil, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil)
        super(semanticId)
        @labelling_dimension = labellingDimension
        @labelling_characteristic_of = labellingCharacteristicOf
        @date = date
        @description = description
        @name = name
        @characteristic_of = characteristicOf
        @dimension = dimension
        self.semanticType = "dfc-b:LabellingCharacteristic"
        registerSemanticProperty("dfc-b:hasLabellingDimension", &method("labelling_dimension")).valueSetter = method("labelling_dimension=")
        registerSemanticProperty("dfc-b:labellingCharacteristicOf", &method("labelling_characteristic_of")).valueSetter = method("labelling_characteristic_of=")
        registerSemanticProperty("dfc-b:date", &method("date")).valueSetter = method("date=")
        registerSemanticProperty("dfc-b:description", &method("description")).valueSetter = method("description=")
        registerSemanticProperty("dfc-b:name", &method("name")).valueSetter = method("name=")
        registerSemanticProperty("dfc-b:characteristicOf", &method("characteristic_of")).valueSetter = method("characteristic_of=")
        registerSemanticProperty("dfc-b:hasDimension", &method("dimension")).valueSetter = method("dimension=")
      end

      Core::SemanticObject.type_registry[SEMANTIC_TYPE] = self
    end
  end
end

File: typescript-connector/src/models/LabellingCharacteristic.ts
// Class from DFC Business Ontology: #LabellingCharacteristic
import { SemanticObject } from "../core/SemanticObject.js";
import type { Concept } from "./Concept.js";

export interface LabellingCharacteristicParams {
  labellingCharacteristicOf?: string;
  date?: string;
  description?: string;
  name?: string;
  characteristicOf?: string;
  hasDimension?: string;
  hasLabellingDimension?: Concept | string;
}
export class LabellingCharacteristic extends SemanticObject {
  static get SEMANTIC_TYPE(): string {
    return "dfc-b:LabellingCharacteristic";
  }

  labellingCharacteristicOf?: string;
  date?: string;
  description?: string;
  name?: string;
  characteristicOf?: string;
  hasDimension?: string;
  hasLabellingDimension?: Concept | string;

  constructor(
    semanticId: string,
    params?: LabellingCharacteristicParams,
  ) {
    super(semanticId);
    this.labellingCharacteristicOf = params?.labellingCharacteristicOf;
    this.date = params?.date;
    this.description = params?.description;
    this.name = params?.name;
    this.characteristicOf = params?.characteristicOf;
    this.hasDimension = params?.hasDimension;
    this.hasLabellingDimension = params?.hasLabellingDimension;
    this.semanticType = LabellingCharacteristic.SEMANTIC_TYPE;
    this.registerSemanticProperty("dfc-b:labellingCharacteristicOf", () => this.labellingCharacteristicOf);
    this.registerSemanticProperty("dfc-b:date", () => this.date);
    this.registerSemanticProperty("dfc-b:description", () => this.description);
    this.registerSemanticProperty("dfc-b:name", () => this.name);
    this.registerSemanticProperty("dfc-b:characteristicOf", () => this.characteristicOf);
    this.registerSemanticProperty("dfc-b:hasDimension", () => this.hasDimension);
    this.registerSemanticProperty("dfc-b:hasLabellingDimension", () => this.hasLabellingDimension);
  }
  static {
    SemanticObject.typeRegistry.set(LabellingCharacteristic.SEMANTIC_TYPE, LabellingCharacteristic);
  }
}
